src/audio/asrc/asrc.c

```diff
@@ -412,6 +412,11 @@ static int asrc_params(struct comp_dev *dev,
 	/* set source/sink_frames/rate */
 	cd->source_rate = sourceb->stream.rate;
 	cd->sink_rate = sinkb->stream.rate;
+	if (cd->sink_rate == 0) {
```

Comment:
if (!cd->sink_rate)?

Reply:
Yep, I'll change.

Off topic I wonder what's the rationale for this Linux kernel rule. The compiler can issue the same compare to zero operation when == 0 seen plus it's more readable in my opinion. Is it to avoid accidental if (variable = 0) {} ?

Reply:
not sure about the rationale but seems like the universally accepted pattern. @lyakh any insights?

Reply:
The just pushed version fixes this.